src/porepy/viz/exporter.py

```diff
@@ -180,6 +182,183 @@ def __init__(
+                are_subdomain_data (bool or list of bool): comparible with file_names; list
+                    of indicators whether the corresponding grid is a subdomain grid;
+                    default is True
+
+        Raises:
+            ValueError if some of the data is not compatible with the supposedly corresponding
+                grid
+        """
+
+        # Aux. method: Inversoe of _toVectorFormat, used to define vector-ranged values
+        def _toScalarFormat(
+            value: np.ndarray, grid: Union[pp.Grid, pp.MortarGrid]
+        ) -> np.ndarray:
+            """
+            Check whether the value array has the right dimension corresponding
+            to the grid size. If possible, translate the value to a scalar-ranged
+            object.
+
+            Args:
+                value (np.ndarray): input array to be converted
+                grid (pp.Grid or pp.MortarGrid): subdomain or interface to which value
+                    is associated to
+
+            Raises:
+                ValueError if the value array is not compatible with the grid
+            """
+            # Make some checks
+            if not value.size % grid.num_cells == 0:
+                # This line will raise an error if node or face data is exported.
+                raise ValueError("The data array is not compatible with the grid.")
+
+            # Apply flatten with the order compatible with _toVectorFormat.
+            return value.flatten("F")
+
+        # Convert file_names to a list
+        if isinstance(file_names, str):
+            file_names = [file_names]
+
+        # Consider each file separately.
+        # Procedure has three steps:
+        # 1. Determine dimensionality and type of grid, addressed by the vtu file.
+        # 2. Check whether the vtu file is compatible with the corresponding grids.
+        # 3. Transfer data from vtu to the mixed-dimensional grid.
+        for i, file_name in enumerate(file_names):
+
+            # Read all data from vtu
+            vtu_data = meshio.read(file_name)
+
+            # 1. Step: Determine dimension of the grid associated to vtu file, and
+            # whether the grid corresponds to a subdomain or interface.
+
+            # Allow for automatic detection from the file name.
+            if kwargs.pop("automatic", True):
+                # Assuming grid_dim and is_mortar are among the stored vtu data
+                # one could utilize those, but this does not have to be the case.
+                # Thus, use naming convention of the Exporter for this.
+
+                # Remove ending '.vtu' from file_name, and decompose into pieces
+                # separated by '_'.
+                file_name_pieces = Path(file_name).stem.split("_")
+
+                # If the last two are numbers, the first one denotes the dimension.
+                assert file_name_pieces[-1].isnumeric()
+                dim_pos = -2 if file_name_pieces[-2].isnumeric() else -1
```

Comment:
You mean that you do not trust the approach here?

The idea here is that when requesting automatic deduction of dimensionality of the grid and its nature (mortar or subdomain), one can set `automatic=True`. Then these properties are deduced purely from the name of the vtu file, assuming the file uses the same naming convention as used for exporting. Of course, if this is not true, the simple deduction it not possible.

Considering the current naming convention for vtu files when exporting with porepy, grid dimensions are always attached as appendix, and when time dependent data is exported, also the time step is added. (Trouble could be introduced when starting to define own naming conventions including for instance iterations). 

Here it is assumed that the name of a vtu file is of type: name_dim.vtu or name_dim_timestep.vtu. Both dim and timestep are numeric (name can be whatever, it can also contain numbers etc) and this is checked.

When automatic deduction is not trusted, one should set `automatic=False` and provide all extra information through keyword arguments. This may be not the prettiest solution, however, my intention was to ease the process for the standard case.

Any other suggestions?

Reply:
I added a comment in the method documentation stating that standard formating of the files is assumed - effectively the files should be created from PorePy. This will be enough for 99% of the cases, and we can't accommodate the final 1%. 

I'm fine with resolving this comment.